pkg/controller/deployment/deployment_controller.go

```diff
@@ -242,6 +276,33 @@ func (d *DeploymentController) reconcileOldRCs(allRCs []*api.ReplicationControll
 	return true, err
 }
 
+func (d *DeploymentController) scaleDownOldRCs(oldRCs []*api.ReplicationController, deployment extensions.Deployment) (bool, error) {
+
+	scaled := false
+
+	for _, rc := range oldRCs {
+		// Scaling not required.
+		if rc.Spec.Replicas == 0 || rc.Status.ObservedGeneration >= rc.Generation && rc.Status.Replicas == 0 {
```

Comment:
I am not sure about looking into ObservedGeneration. Nothing should modify these rcs except the deployment controller.

cc: @nikhiljindal @bgrant0607